starboard/shared/starboard/player/filter/filter_based_player_worker_handler.cc

```diff
@@ -126,11 +129,11 @@ bool FilterBasedPlayerWorkerHandler::Init(
       SB_LOG(ERROR) << "Audio channels requested " << required_audio_channels
                     << ", but currently supported less than or equal to "
                     << supported_audio_channels;
-      OnError(
-          kSbPlayerErrorCapabilityChanged,
+      std::string error_message =
           FormatString("Required channel %d is greater than maximum channel %d",
-                       required_audio_channels, supported_audio_channels));
-      return false;
+                       required_audio_channels, supported_audio_channels);
+      OnError(kSbPlayerErrorCapabilityChanged, error_message);
```

Comment:
After adding HandlerResult and returning it at the end of this function, do we still need to call OnError callback?

Reply:
No, I can remove that to avoid reporting the same message twice.